src/main/js/controls/Timeline/Timeline.js

```diff
@@ -197,6 +197,20 @@ class Timeline extends Construct {
             );
         createDefs(this.config, this.svg);
         createAxes(this.axis, this.scale, this.config, this.svg);
+        const ticks = document.getElementsByClassName("tick");
+        const firstTick = ticks[0].getBoundingClientRect().width;
+        const lastTick = ticks[ticks.length - 1].getBoundingClientRect().width;
+        const xAxis = document
+            .getElementsByClassName("carbon-axis carbon-axis-x")[0]
+            .getBoundingClientRect().width;
+        if (xAxis + firstTick / 2 + lastTick / 2 > this.config.canvasWidth) {
```

Comment:
In the description  you say `Added half of width of first and last ticks as left and right padding if xAxis width is more than canvas width.` 
Why is this necessary?
Why is X Axis width more than canvas width?

Reply:
Actually the sum of X-Axis width and half of first and last ticks becomes more than canvas width.

Reply:
Right, you have a fix here but I am more interested in knowing why the issue is created in the first place so that we can better understand it.

Reply:
Let's say the canvas and X axis both span from 0 to 1200px so the end ticks which start at points 0 and 1200 get cut off because they they have their own widths , so we need some padding values to incorporate these end ticks.

Reply:
So based on what you are saying, if the end ticks werent there then would this be fixed?

Reply:
Yes in certain cases the X axis shortens to an extent that end ticks are not displayed and issue is fixed.